### ISSUE TITLE
Trying to save a blob file in Outlook for macOS using javascript

### ISSUE THREAD

**Issue by Hombr33** [OfficeDev/office-js#503]
Ok so I have a `Blob ` file that I'm trying to save on local disk (the `Blob ` is created locally from an `ArrayBuffer`). The problem I'm facing is that Outlook for macOS does not want to open a `FileSaverDialog` or something similar to let the user save the file.

### My code is working perfectly in Outlook for Web (OWA) on all browsers (including Safari) and Outlook for Windows.

```
const blob = new Blob([body]); //body is an ArrayBuffer created locally
const link = document.createElement('a');
if (link.download !== undefined) {
    const url = URL.createObjectURL(blob);
    link.setAttribute('href', url);
    link.setAttribute('download', fileName);
    link.style.visibility = 'hidden';
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
}
```

**PS:** I have tried using `window.open(blob);`, but Outlook for macOS throws an error in that situation: "_There is no application set to open URL blob:https://..._"

**PS 2:** I have also tried using ```Office.context.ui.displayDialogAsync(url, { height: 100, width: 100, requireHTTPS: true });``` ,but still getting the same error.

How am I supposed to prompt a SaveFileDialog for the user ? Thank you in advance

**Comment by kbrandl** [comment]
@Hombr33 apologies for the lack of response on this; seems like this issue was overlooked until now. Are you still having this problem?

**Comment by Hombr33** [comment]
Hello sir, yes I still have this problem. We have put the project on hold
actually because of this. Help would be much appreciated.

**Comment by kbrandl** [comment]
@Hombr33 thanks for the quick response. @exextoc can you please investigate?

**Comment by AnjanaValsan** [comment]
@Hombr33 apologies for the delay in fixing this. We are working on fixing it. We will update this thread as soon as its done. I should be able to give you an ETA by next week

**Comment by jargil** [comment]
@Hombr33 - we'll investigate and post an update as soon as we have something. Sorry for the delays here.

**Comment by Hombr33** [comment]
Thank you for the investigation. I am looking forward to see the new patch!

Best regards,
Damian Radinoiu

**Comment by jargil** [comment]
@Hombr33 - opening a blob within the Office clients is not supported in Windows or Mac. One possible solution you could pursue in both platforms is to call Office.context.ui.openBrowserWindow(URL); with a URL that you own and from there once the full browser opens, create your blob object and get the browser to download the file.

**Comment by jargil** [comment]
Thanks for filing your bug with us. I'm closing this issue as this is by design.

**Comment by Lilyliu123** [comment]
Is openBrowserWindow function still available? When I use this I got the compile error that openBrowserWindow does not exist.

**Comment by Kashishmsft** [comment]
I could not find Office.context.ui.openBrowserWindow(url) even though I am using latest monthly channel for Office 365. It gave me compile errors in Word Office JS Add-in. 

Any other alternatives available ?

**Comment by plamenbotev** [comment]
@jargil Please note that this functionality was working on windows desktop client. Can you please share any documentation that points when this was removed?
I also can not find openBrowserWindow. I will appreciate any input on that as I currently can not find any work around and we have lost functionality that was working on production and still works for older build of the desktop client.

**Comment by exextoc** [comment]
We have fixed the bug and Office.context.ui.openBrowserWindow(url) should be available in office-js-api version: 20200812.3 or later in macOS.

**Comment by TakeEasyBoy** [comment]
> @Hombr33 - opening a blob within the Office clients is not supported in Windows or Mac. One possible solution you could pursue in both platforms is to call Office.context.ui.openBrowserWindow(URL); with a URL that you own and from there once the full browser opens, create your blob object and get the browser to download the file.

@Hombr33 Hello sir.
I'm now using this solution to solve my problem in add-ins.
But when i use it in an dialog which opened by ```Office.context.ui.displayDialogAsync```API. then i call the ```Office.context.ui.openBrowserWindow```API  to open my own ```download.html``` to trigger download. just like this
![image](https://user-images.githubusercontent.com/31021942/129828188-b60afe7d-da32-4948-90dd-fed9e36253a4.png)
It did nothing, no browser window was opened.
I have printed the ```Office.context.ui``` by  ```console.log```,it gave me like this 
![image](https://user-images.githubusercontent.com/31021942/129827683-3f153688-ed4a-4c56-b481-cc9aff04b8d9.png)

I am using this [Office.js](https://appsforoffice.microsoft.com/lib/1/hosted/Office.js)
I couldn't figure out what cause this issue. So could u help me do some investigation?
Thank u.
